### DISCUSSION TITLE
Default user stanley to be changed

### DISCUSSION THREAD

**Discussion by ManuGeodis** [StackStorm/st2#5757] [Q&A]
I'm planning to integrate stackstorm in my monitoring infrastructure. I have my monitoring infrastructre in Azure VM'S. Satckstorm instance is also running on azure vm. Is it possible to change the default user stanley to any custom name?

**Comment by arm4b** [comment]
@ManuGeodis  Yes, the system user is configurable in `st2.conf`, see https://github.com/StackStorm/st2/blob/eeddc49d3858839da4a911719264748cac5f4c39/conf/st2.conf.sample#L335-L339

**Reply by ManuGeodis** [reply]
I changed the user and key file in st2.conf

[system_user]
user = captain
ssh_key_file = /home/captain/.ssh/captain_rsa


But I'm facing these errors

    "error": "Failed connecting to host 10.223.16.29. q must be exactly 160, 224, or 256 bits long",
    "traceback": "Traceback (most recent call last):\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/runners/parallel_ssh.py\\", line 278, in _connect\
    client.connect()\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/runners/paramiko_ssh.py\\", line 171, in connect\
    self.client = self._connect(host=self.hostname, socket=self.bastion_socket)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/runners/paramiko_ssh.py\\", line 787, in _connect\
    client.connect(**conninfo)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/paramiko/client.py\\", line 435, in connect\
    self._auth(\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/paramiko/client.py\\", line 682, in _auth\
    self._transport.auth_publickey(username, key)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/paramiko/transport.py\\", line 1634, in auth_publickey\
    return self.auth_handler.wait_for_response(my_event)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/paramiko/auth_handler.py\\", line 244, in wait_for_response\
    raise e\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/paramiko/transport.py\\", line 2163, in run\
    handler(self.auth_handler, m)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/paramiko/auth_handler.py\\", line 375, in _parse_service_accept\
    sig = self.private_key.sign_ssh_data(blob, algorithm)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/paramiko/dsskey.py\\", line 109, in sign_ssh_data\
    key = dsa.DSAPrivateNumbers(\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/cryptography/hazmat/primitives/asymmetric/dsa.py\\", line 244, in private_key\
    return backend.load_dsa_private_numbers(self)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/cryptography/hazmat/backends/openssl/backend.py\\", line 826, in load_dsa_private_numbers\
    dsa._check_dsa_private_numbers(numbers)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/cryptography/hazmat/primitives/asymmetric/dsa.py\\", line 282, in _check_dsa_private_numbers\
    _check_dsa_parameters(parameters)\
  File \\"/opt/stackstorm/st2/lib/python3.8/site-packages/cryptography/hazmat/primitives/asymmetric/dsa.py\\", line 274, in _check_dsa_parameters\
    raise ValueError(\\"q must be exactly 160, 224, or 256 bits long\\")\
ValueError: q must be exactly 160, 224, or 256 bits long\
"
  }
}"
  traceback: "  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2actions/container/base.py", line 117, in _do_run
    runner.pre_run()
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/runners/paramiko_ssh_runner.py", line 206, in pre_run
    self._parallel_ssh_client = ParallelSSHClient(**client_kwargs)
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/runners/parallel_ssh.py", line 90, in __init__
    connect_results = self.connect(raise_on_any_error=raise_on_any_error)
  File "/opt/stackstorm/st2/lib/python3.8/site-packages/st2common/runners/parallel_ssh.py", line 131, in connect
    raise NoHostsConnectedToException(msg)


Do we have to update in any other files?

**Reply by arm4b** [reply]
What if you configure user `stanley` with the same ssh key `ssh_key_file`? Will it connect OK or fail in the same way?

Try to make it work with the defaults first, switch the settings next.

**Reply by arm4b** [reply]
Here are the default instructions that used to configure default user stanley and its ssh key:
https://github.com/StackStorm/st2-packages/blob/f2cf7748110c70faa932de0f93aa4e5c755a9933/scripts/st2bootstrap-deb.sh#L256-L297

hope that helps.

**Reply by ManuGeodis** [reply]
@armab The default user stanley is working for me in my test environment. I need to know whether is it possible to change the username from stanley to a custom name that I can use in my production.

Also need to know that Is it possible to run an action from stackstorm using rest api of Bigfix.

**Reply by arm4b** [reply]
Is the user `captain` configured with the instructions above?
Does it use the same SSH key that worked for `stanley`?
Is it the same environment where these 2 users are configured?

**Reply by ManuGeodis** [reply]
@armab  The user captain is configured with the same instructions and it uses a new key that is generated for captain.

Actually there is a small change in my requirement. Since I'm having different servers with different applications in my environment, So using a user either stanley or custom user, We have to give different permissions for this user in different server for running actions. We are using Big Fix for linux configuration management. So i it possible to run stackstorm action using a rest api call to the big fix other than using user?

**Reply by arm4b** [reply]
I can recommend setting up default user and get it working first, then switching to the custom user in the same environment changing one thing at a time while trying to find which step/configuration diff failed.
Custom user functionality is supported. The issue you're having is likely with some configuration.
